tests/test_s3.py

```diff
@@ -43,6 +44,8 @@ def setUp(self):
         # Enable S3 mocking of tests.
         self.mock_s3.start()
 
+        clean_test_environment(self)
```

Comment:
I'm ambivalent as to whether this line needs a comment before it explaining that we do this to be safe against S3_ENDPOINT_URL is set. The function name seems clear enough so as long as we have a docstring in that function that explains its purpose I think it's fine.